models/project/issue.go

```diff
@@ -102,34 +104,38 @@ func MoveIssuesOnProjectBoard(ctx context.Context, board *Board, sortedIssueIDs
 	})
 }
 
-func (b *Board) moveIssuesToAnotherColumn(ctx context.Context, columnID int64) error {
-	if b.ID == columnID {
+func (b *Board) moveIssuesToAnotherColumn(ctx context.Context, newColumn *Board) error {
+	if b.ID == newColumn.ID {
 		return nil
 	}
-	_, err := db.GetEngine(ctx).Exec("UPDATE `project_issue` SET project_board_id = ? WHERE project_board_id = ? ", columnID, b.ID)
+	if b.ProjectID != newColumn.ProjectID {
+		return fmt.Errorf("columns have to be in the same project")
+	}
+	_, err := db.GetEngine(ctx).Exec("UPDATE `project_issue` SET project_board_id = ? WHERE project_board_id = ? ", newColumn.ID, b.ID)
 	return err
 }
 
-// MoveColumnsOnProject moves or keeps issues in a column and sorts them inside that column
+// MoveColumnsOnProject moves or keeps column in a project and sorts them inside that column
```

Comment:
The second part no longer makes sense

Reply:
833ec563792b2fa883dbd147df19bb09bef61277